### PR TITLE
asserts: basic support for validation assertion and refresh-control

### PR DESCRIPTION
**Pull request by pedronis**
This adds the new validation assertion for refresh-control and support for the "refresh-control" header in snap-declaration used to list gated snaps.

### CODE REVIEW

File: asserts/snap_asserts.go [hunk 1]
```diff
@@ -365,7 +365,7 @@ func (validation *Validation) checkConsistency(db RODatabase, acck *AccountKey)
 		"snap-id": validation.ApprovedSnapID(),
 	})
 	if err == ErrNotFound {
-		return fmt.Errorf("validation assertion for snap-id %q does not have a matching snap-declaration assertion", validation.SnapID())
+		return fmt.Errorf("validation assertion for snap-id %q does not have a matching snap-declaration assertion for it", validation.ApprovedSnapID())
```
Comment (emgee) [inline asserts/snap_asserts.go]: Personally, I don't think the "for it" adds much to the message. Ditto below.

Comment (emgee) [inline asserts/snap_asserts.go]: I see & like what you're trying to do with the "for" v "by" in these error messages but it seems a bit subtle. Maybe this message should be "validation assertion for snap-id {{snap-id}} does not have a matching snap-declaration assertion for approved-snap-id {{approved-snap-id}}"?

File: asserts/snap_asserts_test.go [hunk 2]
```diff
@@ -735,7 +735,43 @@ func (vs *validationSuite) TestRevocation(c *C) {
...
+		"series: 16\n" +
+		"snap-id: snap-id-1\n" +
+		"approved-snap-id: snap-id-2\n" +
+		"approved-snap-revision: 42\n" +
+		"revoked: false\n" +
+		"revision: 1\n" +
+		vs.tsLine +
+		"sign-key-sha3-384: Jv8_JiHiIzJVcO9M55pPdqSDWUvuhfDIBJUS-3VW7F_idjix7Ffn5qMxB21ZQuij" +
+		"\n\n" +
+		"AXNpZw=="
```
Comment (emgee) [inline asserts/snap_asserts_test.go]: Reuse makeValidEncoded here?

File: asserts/snap_asserts_test.go [hunk 3]
```diff
@@ -735,7 +735,43 @@ func (vs *validationSuite) TestRevocation(c *C) {
...
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(err, IsNil)
+	validation := a.(*asserts.Validation)
+	c.Check(validation.Revoked(), Equals, false)
+}
+
+func (vs *validationSuite) TestRevokedInvalid(c *C) {
```
Comment (emgee) [inline asserts/snap_asserts_test.go]: I think we could sqaush this test into TestDecodeInvalid?

Comment (pedronis) [inline asserts/snap_asserts_test.go]: I thought of that but there's no revoked there to tweak

File: asserts/snap_asserts_test.go (hunk 2 above)
Comment (pedronis) [inline asserts/snap_asserts_test.go]: @emgee same issue, not revoked there

**Comment by pedronis** [comment]
picks up #1748

**Comment by emgee** [comment]
+1

**Comment by pedronis** [comment]
picking up #1837 too

**Pull request merged**